independent-projects/tools/analytics-common/src/main/java/io/quarkus/analytics/rest/RestClient.java

```diff
@@ -0,0 +1,154 @@
+package io.quarkus.analytics.rest;
+
+import static io.quarkus.analytics.util.StringUtils.getObjectMapper;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.util.Base64;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+import io.quarkus.analytics.dto.config.AnalyticsRemoteConfig;
+import io.quarkus.analytics.dto.config.Identity;
+import io.quarkus.analytics.dto.config.RemoteConfig;
+import io.quarkus.analytics.dto.segment.Track;
+import io.quarkus.devtools.messagewriter.MessageWriter;
+
+/**
+ * Client to post the analytics data to the upstream collection tool.
+ * We use plain REST API calls and not any wrapping library.
+ */
+public class RestClient implements ConfigClient, SegmentClient {
+
+    static final String IDENTITY_ENDPOINT = "v1/identify";
+    static final String TRACK_ENDPOINT = "v1/track";
+    static final int TIMEOUT = 1000;// milliseconds
+    public static final int TTL = TIMEOUT * 2;// milliseconds
+    static final URI CONFIG_URI = getUri(
+            "https://raw.githubusercontent.com/brunobat/tests/main/03656937-19FD-4C83-9066-C76631D445EA");//FIXME config location
```

Comment:
Should this and the auth-header be configurable?

Reply:
In the intellij plugin it's not... 
And this is not the final location. It must live in the Quarkus repo. I'll update it in a follow up PR along with the segment prod key.